matrix_sdk/src/room/joined.rs

```diff
@@ -0,0 +1,520 @@
+    pub async fn ban_user(&self, user_id: &UserId, reason: Option<&str>) -> Result<()> {
+        let request = assign!(ban_user::Request::new(self.inner.room_id(), user_id), {
+            reason
+        });
+        self.client.send(request, None).await?;
+        Ok(())
+    }
+
+    /// Kick a user out of this room.
+    ///
+    /// # Arguments
+    ///
+    /// * `user_id` - The `UserId` of the user that should be kicked out of the room.
+    ///
+    /// * `reason` - Optional reason why the room member is being kicked out.
+    pub async fn kick_user(&self, user_id: &UserId, reason: Option<&str>) -> Result<()> {
+        let request = assign!(kick_user::Request::new(self.inner.room_id(), user_id), {
+            reason
+        });
+        self.client.send(request, None).await?;
+        Ok(())
+    }
+
+    /// Invite the specified user by `UserId` to this room.
+    ///
+    /// # Arguments
+    ///
+    /// * `user_id` - The `UserId` of the user to invite to the room.
+    pub async fn invite_user_by_id(&self, user_id: &UserId) -> Result<()> {
+        let recipient = InvitationRecipient::UserId { user_id };
+
+        let request = invite_user::Request::new(self.inner.room_id(), recipient);
+        self.client.send(request, None).await?;
+
+        Ok(())
+    }
+
+    /// Invite the specified user by third party id to this room.
+    ///
+    /// # Arguments
+    ///
+    /// * `invite_id` - A third party id of a user to invite to the room.
+    pub async fn invite_user_by_3pid(&self, invite_id: Invite3pid<'_>) -> Result<()> {
+        let recipient = InvitationRecipient::ThirdPartyId(invite_id);
+        let request = invite_user::Request::new(self.inner.room_id(), recipient);
+        self.client.send(request, None).await?;
+
+        Ok(())
+    }
+
+    /// Send a request to notify this room of a user typing.
+    ///
+    /// # Arguments
+    ///
+    /// * `typing` - Whether the user is typing, and how long.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// # use std::time::Duration;
+    /// # use matrix_sdk::{
+    /// #     Client, SyncSettings,
+    /// #     api::r0::typing::create_typing_event::Typing,
+    /// #     identifiers::room_id,
+    /// # };
+    /// # use futures::executor::block_on;
+    /// # use url::Url;
+    /// # block_on(async {
+    /// # let homeserver = Url::parse("http://localhost:8080").unwrap();
+    /// # let mut client = Client::new(homeserver).unwrap();
+    /// # let room_id = room_id!("!test:localhost");
+    /// # let room = client
+    /// #    .get_joined_room(&room_id!("!SVkFJHzfwvuaIEawgC:localhost"))
+    /// #    .unwrap();
+    /// # room
+    ///     .typing_notice(Typing::Yes(Duration::from_secs(4)))
+    ///     .await
+    ///     .expect("Can't get devices from server");
+    /// # });
+    ///
+    /// ```
+    pub async fn typing_notice(&self, typing: impl Into<Typing>) -> Result<()> {
+        let user_id = self
```

Comment:
The `Room` struct contains an `own_user_id` field over [here](https://matrix-org.github.io/matrix-rust-sdk/matrix_sdk/struct.Room.html#method.own_user_id). Which removes the `Option` and the await point.